packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -435,22 +435,25 @@ class _CupertinoPageTransitionState extends State<CupertinoPageTransition> {
     super.didUpdateWidget(oldWidget);
     if (oldWidget.primaryRouteAnimation != widget.primaryRouteAnimation
     || oldWidget.secondaryRouteAnimation != widget.secondaryRouteAnimation
-    || oldWidget.child != widget.child || oldWidget.linearTransition != widget.linearTransition) {
```

Comment:
Is this related to the memory leak? If not, maybe this should be a separate change.

Reply:
Well now this PR is not about memory leaks anymore since https://github.com/flutter/flutter/pull/147133 got merged in between. Maybe I should rename this PR. Would "Fix jank in cupertino page transition" sounds nice to you ?